components/FilterForm/index.js

```diff
@@ -30,10 +54,18 @@ export default function FilterForm({ plants, onAddPreference }) {
   function handleSubmit(event) {
     event.preventDefault();
 
-    const newPreference = {
-      preferenceTitle: event.target.elements.title.value,
+    const filterSettings = {
+      plantSize,
+      sunlightRequirement,
+      waterNeeds,
+      optimalTemperature,
+      petFriendly,
+    };
+
+    const preferenceData = {
+      preferenceTitle: preferenceTitle,
       preferencePlants: plants
-        .filter(
+        ?.filter(
```

Comment:
this pattern often indicates problems elsewhere, if plants are undefined at some point, you could use an early return to not render this component at all, if there are no plants yet.

This could also help your useState initialization I guess